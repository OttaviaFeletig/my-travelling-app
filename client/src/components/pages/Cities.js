import React, { Component } from 'react'

export default class Cities extends Component {
  render() {
    return (
      <div>
        <p>This is the cities page</p>
      </div>
    )
  }
}
