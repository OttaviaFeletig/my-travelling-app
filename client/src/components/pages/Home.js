import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { googleAuth } from "../../actions/usersAction";

// import Slider from 'react-animated-slider'
// import 'react-animated-slider/build/horizontal.css'
import "../../style/Home.css";

import arrow from "../../icons/arrow.png";

export class Home extends Component {
  componentDidMount() {
    console.log(this.props.location);
    console.log(this.props.location.search);
    const url = this.props.location.search;
    if (url !== "") {
      const code = url.split("=")[1];

      this.props.googleAuth(code);
      console.log(code);
    }
  }
  render() {
    return (
      <div className="home">
        <Link style={linkStyle} to="/cities">
          <img style={arrowStyle} src={arrow} alt="arrow" />
        </Link>
      </div>
    );
  }
}

const arrowStyle = {
  height: "30%"
};

const linkStyle = {
  height: "100vh",
  display: "flex",
  alignContent: "center",
  justifyContent: "center",
  alignItems: "center"
};

const mapDispatchToProps = dispatch => {
  return {
    googleAuth: code => dispatch(googleAuth(code))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(Home);
