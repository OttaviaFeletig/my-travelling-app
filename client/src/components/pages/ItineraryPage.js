import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchItineraries } from '../../actions/itineariesAction';

class ItineraryPage extends Component {
  componentDidMount() {
    this.props.dispatch(fetchItineraries(this.props.match.params.id))
  } 

  render() {
    console.log(this.props.itineraries)
    return (
      <div>
        
      </div>
    )
  }
}
const mapStateProps = (state) => {
  return {
    itineraries: state.itineraries,
    loading: state.loading,
    error: state.error
  }
}

ItineraryPage.propTypes = {
  itineraries: PropTypes.object.isRequired
}
export default connect(mapStateProps)(ItineraryPage)
