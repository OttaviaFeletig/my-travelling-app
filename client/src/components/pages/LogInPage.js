import React, { Component } from 'react'
import { Link } from 'react-router-dom'


export default class LogInPage extends Component {
  render() {
    
    return (
      <div style={logInPageStyle}>
        <p>You don't have an account? Sign up <Link to='/signUp'>here</Link></p>
      </div>
    )
  }
}

const logInPageStyle = {
  paddingTop: '80px'
}
