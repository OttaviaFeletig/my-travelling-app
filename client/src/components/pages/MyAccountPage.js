import React, { Component } from 'react'

export default class MyAccountPage extends Component {
  render() {
    return (
      <div style={myAccountPageStyle}>
        <p>this is my account</p>
      </div>
    )
  }
}

const myAccountPageStyle = {
    paddingTop: '200px',
    margin: '0 20px 0 20px'
  }
