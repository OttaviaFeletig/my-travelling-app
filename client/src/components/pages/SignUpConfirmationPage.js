import React, { Component } from 'react';
import { Link } from 'react-router-dom'

export default class SignUpConfirmationPage extends Component {

    
  render() {

    const {username, email, avatarPicture} = this.props.location.state.detail;

    return (
      <div style={confPageStyle}>
        <h3>Thank you for registering!</h3>
        <p>You submitted the following information:</p>
        <img style={avatarStyle} src={avatarPicture} alt='avatar'></img>
        <ul>
            <li>Username: {username}</li>
            <li>Email: {email}</li>
        </ul>
        <Link to='login'>Click here to log in!</Link>
      </div>
    )
  }
}

const confPageStyle = {
    paddingTop: '80px',
    margin: '0 20px 0 20px'
} 
const avatarStyle = {
    borderRadius: '50%',
    width: '50%'
}
