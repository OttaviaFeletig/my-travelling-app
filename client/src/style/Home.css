.home{
    padding-top: 30px;
}
.slider{
    height: 650px;
}
.slide{
    width: 100%;
    height: 100%;
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-around
}
polygon{
    fill: #32607F;
}