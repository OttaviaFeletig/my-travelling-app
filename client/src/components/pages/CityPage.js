import React, { Component } from 'react';
import { connect } from 'react-redux';
import {fetchCities} from '../../actions/citiesAction';
import Cities from '../functional_component/Cities';


class CityPage extends Component {

  componentDidMount() {
    this.props.dispatch(fetchCities())
  } 

  render() {
    //destructuring state
    //it can also be written as const cities = this.state.cities
    const { cities, error, loading } = this.props.cities
    console.log(cities)
    console.log(error)
    console.log(loading)
    if(error) {
        return <div>Error! { error.message }</div>
    }
    if(loading) {
      return <div>Loading...</div>
    }
    return (
      <div style={citiesStyle}>
        <Cities cities={cities} />
      </div>
    )
  }
}
const citiesStyle = {
  paddingTop: '100px'
}

const mapStateProps = (state) => {
  return {
    cities: state.cities,
    loading: state.loading,
    error: state.error
  }
}

export default connect(mapStateProps)(CityPage)
