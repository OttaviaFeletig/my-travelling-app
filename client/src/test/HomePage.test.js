import React from "react";
import { shallow, render } from "enzyme";
import { Home } from "../components/pages/Home";
import { BrowserRouter as Router } from "react-router-dom";
require("../setupTest");

it("Home component snapshot", () => {
  const wrapper = render(
    <Router>
      <Home />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
