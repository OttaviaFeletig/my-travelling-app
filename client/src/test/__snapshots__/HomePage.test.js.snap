// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "home",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "href": "/cities",
          "style": "height:100vh;display:flex;align-content:center;justify-content:center;align-items:center",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "alt": "arrow",
              "src": "arrow.png",
              "style": "height:30%",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "alt": undefined,
              "src": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "alt": undefined,
              "src": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "a",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "href": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "href": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
